tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -329,20 +355,57 @@ private void AssertAuthenticationFails(MongoClientSettings settings)
         {
             return CreateDatabaseUser(client, "admin", userName, password, "readWriteAnyDatabase", mechanisms);
         }
-        
+
         private BsonDocument CreateDatabaseUser(
             MongoClient client,
             string source,
-            string userName, 
-            string password, 
-            string role, 
+            string userName,
+            string password,
+            string role,
             params string[] mechanisms)
         {
             var createUserCommand =
-                $"{{createUser: '{userName}', pwd: '{password}'," 
+                $"{{createUser: '{userName}', pwd: '{password}',"
                 + $"roles: ['{role}'], mechanisms: {mechanisms.ToJson()}}}";
             return client.GetDatabase(source).RunCommand<BsonDocument>(createUserCommand);
         }
-        
+
+        private BsonDocument CreateX509DatabaseUser(MongoClient client, string userName)
+        {
+            var createUserCommand =
+                $"{{ createUser: '{userName}'," +
+                @"   roles: [ { role: 'readWrite', db: 'test' },
+                              { role: 'userAdminAnyDatabase', db: 'admin' } 
+                    ]
+                  }";
+            return client.GetDatabase("$external").RunCommand<BsonDocument>(createUserCommand);
+        }
+
+        private List<object> DropDatabaseUser(MongoClient client, string userName, IEnumerable<string> databases)
+        {
+            var dropUserCommand = $"{{ dropUser: '{userName}'}}";
+            
+            return databases.Select(TryDropUser).ToList();
+
+            object TryDropUser(string database)
+            {
+                object result;
```

Comment:
This is a bit unusual, but this allows us to iterate through all the dbs in `databases` and store the result/exception and return it. We expect an exception in the case of the user not existing, which will be the case on the first framework we test but not for the other frameworks.

We could roll our own `Either` class so that we could return an `Either<BsonDocument, MongoException>` data structure, but as this is just test code, I don't mind using `object` so that we can use the debugger to examine the response or the Exception if needed. This method has the added benefit in that when multiple databases are passed in, we try to drop the user on all the databases, even if the command throws an exception on one of them. And the only time we need to examine the list returned, as of now, will be during local debugging.

Reply:
Is there any consumer of the result of `DropDatabaseUser`? I guess the answer is no. if so the result of DropDatabaseUser is not used.
I would say if you need this information only for debugging, in any case, you will need to make code changes to see them, at least because there is no variable to assign the result of this method. So, adding a return part for `DropDatabaseUser` method can be also added only in local code.

Reply:
So, how about something like this:

    TryDrop(out exception)
?

Reply:
I think, leaving any not used code(even in tests) is not a good sign and should be avoided (maybe unless it's too complicated to do it each time). But we can discuss it.

Reply:
What if we stored the result in the integration test so that no changes would be needed when examining the test in a debugger? That way the return value is used, and no changes will be needed when debugging? I've made this change in my latest commit.

Reply:
Something somewhat amusing I just noticed: C# itself uses `paramName: value` for its named parameters but we prefer `key : value` for BsonDocuments :)

Reply:
> 
> 
> What if we stored the result in the integration test so that no changes would be needed when examining the test in a debugger? That way the return value is used, and no changes will be needed when debugging? I've made this change in my latest commit.

sorry, I still prefer to avoid logic for local tests in final code, but I will be ok if you and other reviewers will be also ok with it.

Reply:
Gotcha, we'll see what @rstam says then :)

I will note that we have some precedence for returning things we don't care about in this file e.g. the other user creation methods return a `BsonDocument` that we never check.

Reply:
See comment on line 425.